sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cosmos/AzureCosmosProperties.java

```diff
@@ -0,0 +1,191 @@
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.DeprecatedConfigurationProperty;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.constraints.NotEmpty;
+import java.util.regex.Pattern;
+
+/**
+ * Configuration properties for Cosmos database, consistency, telemetry, connection, query metrics and diagnostics.
+ */
+@Validated
+@ConfigurationProperties(AzureCosmosProperties.PREFIX)
+public class AzureCosmosProperties extends AzureProperties implements InitializingBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureCosmosProperties.class);
+    public static final String PREFIX = "spring.cloud.azure.cosmos";
+    public static final String URI_REGEX = "http[s]{0,1}://.*.documents.azure.com.*";
+
+    @NotEmpty
+    private String uri;
+
+    @NotEmpty
+    private String key;
+
+    /**
+     * Override enabled, session capturing is enabled by default for {@link ConsistencyLevel#SESSION}
+     */
+    private boolean sessionCapturingOverrideEnabled;
+    private boolean connectionSharingAcrossClientsEnabled;
+    private boolean contentResponseOnWriteEnabled;
+    private CosmosPermissionProperties permissions;
+    private GatewayConnectionConfig gatewayConnectionConfig;
+    private DirectConnectionConfig directConnectionConfig;
+
+    /**
+     * Document DB consistency level.
+     */
+    private ConsistencyLevel consistencyLevel;
+
+    /**
+     * Document DB database name.
+     */
+    @NotEmpty
+    private String database;
+
+    /**
+     * Populate Diagnostics Strings and Query metrics
+     */
+    private boolean populateQueryMetrics;
+
+    /**
+     * Whether allow Microsoft to collect telemetry data.
+     */
+    private boolean allowTelemetry = true;
+
+    /**
+     * Represents the connection mode to be used by the client in the Azure Cosmos DB database service.
+     */
+    private ConnectionMode connectionMode;
+
+    @Override
+    public void afterPropertiesSet() {
+        validateUri();
+    }
+
+    /**
+     * Response Diagnostics processor
+     * Default implementation is to log the response diagnostics string
+     */
+    private ResponseDiagnosticsProcessor responseDiagnosticsProcessor =
+        responseDiagnostics -> {
+            if (populateQueryMetrics) {
+                LOGGER.info("Response Diagnostics {}", responseDiagnostics);
+            }
+        };
+
+
+    public String getDatabase() {
+        return database;
+    }
+
+    public void setDatabase(String databaseName) {
+        this.database = databaseName;
+    }
+
+    public ConsistencyLevel getConsistencyLevel() {
+        return consistencyLevel;
+    }
+
+    public void setConsistencyLevel(ConsistencyLevel consistencyLevel) {
+        this.consistencyLevel = consistencyLevel;
+    }
+
+    @Deprecated
+    @DeprecatedConfigurationProperty(
```

Comment:
do we still need to keep this deprecated property in the 4.0 release?

Reply:
No, we don't.